src/snowflake/cli/api/config.py

```diff
@@ -47,6 +48,50 @@ class Empty:
 )
 
 
+@dataclass
+class ConnectionConfig:
+    account: Optional[str] = None
+    user: Optional[str] = None
+    password: Optional[str] = None
+    host: Optional[str] = None
+    region: Optional[str] = None
+    port: Optional[int] = None
+    database: Optional[str] = None
+    schema: Optional[str] = None
+    warehouse: Optional[str] = None
+    role: Optional[str] = None
+    authenticator: Optional[str] = None
+    private_key_path: Optional[str] = None
```

Comment:
WDY about using pydantic model?

Reply:
I thought about it but I wanted to make this change as small as possible for purpose of the patch version and then continue refactoring on the main branch (the same as changes mentioned in https://github.com/snowflakedb/snowflake-cli/pull/909#discussion_r1528699025).